src/app/views/homepage/edit/EditHomepageController.jsx

```diff
@@ -91,25 +91,30 @@ export class EditHomepageController extends Component {
         collections
             .getContentCollectionDetails(this.props.params.collectionID)
             .then(collection => {
-                const lastEventIndex = collection.eventsByUri["/data.json"].length - 1;
-                const lastEvent = collection.eventsByUri["/data.json"][lastEventIndex];
-                let collectionState = "";
-
-                switch (lastEvent.type) {
-                    case "COMPLETED":
-                        collectionState = "complete";
-                        break;
-                    case "EDITED":
-                        collectionState = "inProgress";
-                        break;
-                    default:
-                        collectionState = "reviewed";
+                const pageHasEvent = collection.hasOwnProperty("eventsByUri") && collection.eventsByUri.hasOwnProperty("/data.json");
+                if (pageHasEvent) {
```

Comment:
In the case there is no page event what would happen? Should we set a default state? Throw an error and prevent editing and instruct user to refresh? Only allow save and preview? 

Happy to discuss

Reply:
Have updated the form disabling logic so that it is disabled by default. The form is then enabled if all data is successfully fetched